test/extended/apiserver/resource_quota.go

```diff
@@ -0,0 +1,60 @@
+package apiserver
+
+import (
+	"context"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer][Serial]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("apiserver-resource-quota")
+
+	g.It("The number of created persistent volume claims can not exceed the limitation", func() {
+		testProject := oc.SetupProject()
+		testResourceQuotaName := "my-resource-quota"
+		clusterAdminKubeClient := oc.AdminKubeClient()
+
+		rq := &corev1.ResourceQuota{
+			ObjectMeta: metav1.ObjectMeta{Name: testResourceQuotaName, Namespace: testProject},
+			Spec: corev1.ResourceQuotaSpec{
+				Hard: corev1.ResourceList{
+					"persistentvolumeclaims": resource.MustParse("1"),
+				},
+			},
+		}
+
+		pvc := &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "myclaim-1",
```

Comment:
This should be `generateName` to make sure we're not creating the same PVC twice maybe?

Reply:
@vrutkovs Updated